vllm/entrypoints/openai/api_server.py

```diff
@@ -426,9 +430,10 @@ async def create_completion(request: CompletionRequest, raw_request: Request):
             top_k=request.top_k,
             stop=request.stop,
             ignore_eos=request.ignore_eos,
-            max_tokens=request.max_tokens,
+            max_tokens=request.max_tokens if not echo_self else 1,
```

Comment:
I believe `max_tokens` is not in conflict with `echo`, so this `if` should not exist. We should be able to generate many tokens while returning the logits of the prompt.

Reply:
This is a use case myself encounters; OpenAI API supports echoing without generating any tokens while in vLLM we assume that we have to generate at least 1 token. This is a workaround to support the same semantics --- asking the engine to generate 1 token and set the `echo_self` flag to true and removing the generated token at the end before sending back the response.

Reply:
Correct me if I'm wrong, but I think what this line does is when `echo` is on, `max_tokens` have to be 1?

Reply:
Nope, it means when `echo==True and max_tokens==0`. Please see line 379.

Reply:
got it. Thanks!